purchases/src/main/kotlin/com/revenuecat/purchases/errors.kt

```diff
@@ -67,4 +67,8 @@ enum class PurchasesErrorCode(val code: Int, val description: String) {
         36,
         "Request failed signature verification. Please see https://rev.cat/trusted-entitlements for more info.",
     ),
+    SyncingAttributesRateLimitReached(
+        37,
```

Comment:
Is this error number correct? Note that it needs to match iOS so it's the same in the hybrids.

Reply:
I think error number 37 already exists in iOS and is equivalent to number 36 on iOS so we already have a mismatch.

Reply:
It looks like everything is off-by-one between iOS and Android error codes? 🤔

Reply:
I think we just skip error numbers in one platform if it doesn't apply e.g. there is no error 26 `productDiscountMissingIdentifierError` in Android, but there is in iOS.

Reply:
Ugh, it's broken yes. `SignatureVerificationError` should be 37 in Android. I would say for simplicity  make `SyncingAttributesRateLimitReached` 40 in both platforms and we fix `SignatureVerificationError` and `featureNotAvailableInCustomEntitlementsComputationMode` later by making them 38 and 39 🤷‍♂️ 

Btw, I would love to bring this up in the next SDK guild meeting because it's one of the most bug prone thing in the SDKs we have, and I am not sure how to solve it

Reply:
@vegaro Make it a shared CI step that compares and fails if different? 🤷‍♂️

Reply:
Yes, we are going to have to do something on CI, a test. I think that's the only option. Maybe in PHC? But that would too late, after releasing the natives. 

Let's talk about it yes haha